File: Atcoder_#348/ruby/b_farthest_point.rb
n = gets.to_i
points = []

n.times do
  x, y = gets.split.map(&:to_i)
  points << [x, y]
end

def euclidean_distance(point1, point2)
  Math.sqrt((point1[0] - point2[0])**2 + (point1[1] - point2[1])**2)
end

(0...n).each do |i|
  max_distance = -1
  max_point_index = -1

  (0...n).each do |j|
    next if i == j

    distance = euclidean_distance(points[i], points[j])
    if distance > max_distance || (distance == max_distance && j < max_point_index)
      max_distance = distance
      max_point_index = j
    end
  end

  puts max_point_index + 1
end

File: Atcoder_#348/javascript/b_farthest_point.js
function main(input) {
  const lines = input.trim().split('\n');
  const n = parseInt(lines[0], 10);
  const points = lines.slice(1).map(line => {
    const [x, y] = line.split(' ').map(Number);
    return { x, y };
  });

  function euclideanDistance(point1, point2) {
    return Math.sqrt((point1.x - point2.x) ** 2 + (point1.y - point2.y) ** 2);
  }

  let result = '';

  for (let i = 0; i < n; i++) {
    let maxDistance = -1;
    let maxPointIndex = -1;

    for (let j = 0; j < n; j++) {
      if (i === j) continue;

      const distance = euclideanDistance(points[i], points[j]);
      if (distance > maxDistance || (distance === maxDistance && j < maxPointIndex)) {
        maxDistance = distance;
        maxPointIndex = j;
      }
    }

    result += (maxPointIndex + 1) + '\n';
  }

  console.log(result.trim());
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));
